src/common/hwio_buddy_2209_02.cpp

```diff
@@ -532,14 +535,22 @@ void digitalWrite(uint32_t marlinPin, uint32_t ulVal) {
 #ifdef SIM_HEATER_BED_ADC
         sim_bed_set_power(ulVal ? 100 : 0);
 #else //SIM_HEATER_BED_ADC
-        _hwio_pwm_analogWrite_set_val(HWIO_PWM_HEATER_BED, ulVal ? _pwm_analogWrite_max[HWIO_PWM_HEATER_BED] : 0);
+    {
+        const uint32_t value = ulVal ? _pwm_analogWrite_max[HWIO_PWM_HEATER_BED] : 0;
+        _hwio_pwm_analogWrite_set_val(HWIO_PWM_HEATER_BED, value);
```

Comment:
As there is more places where the bed/nozzle pwm is being set, wouldn't it be cleaner to record the metric within `_hwio_pwm_analogWrite_set_val`?

Reply:
Yeah, this makes sense.

Reply:
fixed